client/lib/domains/dns/reducer.js

```diff
@@ -46,6 +44,36 @@ function deleteDns( state, domainName, record ) {
 	return update( state, command );
 }
 
+function markDnsForDeletion( state, domainName, record, isBeingDeleted ) {
+	const index = findDns( state, domainName, record ),
```

Comment:
I was torn on this - I agree that it's more readable, but on the other hand it's less "strict" - you can pass any object, without the `isBeingDeleted` property. On the _other_ hand, it's an internal function, so that shouldn't be an issue (not sure if it's at all :stuck_out_tongue:), so I'll go with your suggestion :+1: